app/models/metric.rb

```diff
@@ -3,7 +3,21 @@ class Metric < ApplicationRecord
 
   include Metric::Common
 
-  def self.reindex_table_name
-    "metrics_#{Time.now.utc.hour + 1}"
+  def self.table_name
+    Thread.current[:metric_table] || super
+  end
+
+  def self.with_table_for_time(hour = Time.now.utc.hour)
+    Thread.current[:metric_table] = reindex_table_name(hour)
```

Comment:
Can't give a better suggestion here, but with this affecting whether either one or "all" of the tables gets truncated, this seems super scary to me doing this in a `Thread` var (and this is coming from someone who used thread variables as well to do stuff in `ManageIQPerformance`).

Just for slight piece of mind, can we make this variable name more verbose, something like `:targeted_metric_subtable_name`, just it avoids the possibility of it being collided by some other code messing with `Thread` variables and possibly re-using that indentifier.